library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -0,0 +1,198 @@
+    #
+    # @param description [String] Snapshot's description.
+    # @return [FsSnapshot] The created snapshot.
+    #
+    # @see FsSnapshot.create
+    def self.create_pre(description)
+      create(:pre, description)
+    end
+
+    # Creates a new 'post' snapshot
+    #
+    # Each 'post' snapshot corresponds with a 'pre' one. If the number of the
+    # 'pre' snapshot is not given, it will try to figure out that number.
+    #
+    # @param description     [String] Snapshot's description.
+    # @param previous_number [Fixnum] Number of the previous snapshot
+    # @return [FsSnapshot] The created snapshot.
+    #
+    # @see FsSnapshot.create
+    def self.create_post(description, previous_number = nil)
+      previous =
+        if previous_number.nil?
+          last = all.reverse.find { |s| s.snapshot_type == :pre || s.snapshot_type == :post }
+          (last && last.snapshot_type == :pre) ? last : nil
+        else
+          find(previous_number)
+        end
+
+      if previous
+        create(:post, description, previous)
+      else
+        log.error "Previous filesystem snapshot was not found"
+        raise PreviousSnapshotNotFound
+      end
+    end
+
+    # Creates a new snapshot
+    #
+    # It raises and exception if Snapper is not configured or if snapshot
+    # creation fails.
+    #
+    # @param snapshot_type [Symbol]    Snapshot's type: :pre, :post or :single.
+    # @param description   [String]    Snapshot's description.
+    # @param previous      [FsSnashot] Previous snapshot.
+    # @return [FsSnapshot] The created snapshot if the operation was
+    #                      successful.
+    private_class_method def self.create(snapshot_type, description, previous = nil)
+      raise SnapperNotConfigured unless configured?
+
+      cmd = CREATE_SNAPSHOT_CMD % [snapshot_type, description]
+      cmd << " --pre-num %s" % previous.number unless previous.nil?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), cmd)
+      if out["exit"] == 0
+        find(out["stdout"].to_i) # The CREATE_SNAPSHOT_CMD returns the number of the new snapshot.
+      else
+        log.error "Snapshot could not be created: #{cmd} returned: #{out}"
+        raise SnapshotCreationFailed
+      end
+    end
+
+    # Returns all snapshots
+    #
+    # It raises and exception if Snapper is not configured.
+    #
+    # @return [Array<FsSnapshot>] All snapshots that exist in the system.
+    def self.all
+      raise SnapperNotConfigured unless configured?
+
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), LIST_SNAPSHOTS_CMD)
+      lines = out["stdout"].lines.grep(VALID_LINE_REGEX) # relevant lines from output.
+      log.info("Retrieving snapshots list: #{LIST_SNAPSHOTS_CMD} returned: #{out}")
+      lines.map do |line|
+        data = line.split("|").map(&:strip)
+        begin
+          timestamp = DateTime.parse(data[3])
+        rescue ArgumentError
+          timestamp = nil
```

Comment:
please at least log data that is not valid DateTime, otherwise we have later trouble to recognize that parsing date failed. ( I know we log complete list, but specific date entry can help with debugging.